CMakeLists.txt

```diff
@@ -207,6 +207,11 @@ if(__INTEGRATE_OPENCL)
     endif()
 endif()
 
+if(BUILD_CPP_TEST AND MSVC)
+  # Use /MT flag to statically link the C runtime
+  set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+endif()
```

Comment:
Not sure what the default should be for LightGBM when it's not built with gtest. For now, I kept the current behavior, which is to specify `/MT` when gtest is enabled.

Reply:
I agree with this, let's keep the current behavior.

And for now, could you please put this down where the existing string-replacement code is? For as long as the testing context is the only place where we explicitly set this, I'd prefer to keep this inside the big `if(BUILD_CPP_TEST)` block, alongside most of the other test-specific configuration.